R/input-switch.R

```diff
@@ -78,16 +78,44 @@ toggle_switch <- function(id, value = NULL, session = get_current_session()) {
 }
 
 # Keep this internal for now until we agree on the UX
-input_dark_mode_switch <- function(id, label, ..., width = "auto") {
-  res <- input_switch(id, label, ..., width = width)
-  # TODO: .navbar should probably do this for .shiny-input-container
-  res <- tagAppendAttributes(res, style = "margin-bottom: 0;")
-  tagAppendAttributes(
-    res, .cssSelector = "input",
-    onclick = "document.documentElement.setAttribute('data-bs-theme', this.checked ? 'dark' : 'light'); $(window).resize();"
+input_dark_mode_switch <- function(id, ..., mode = NULL) {
+  if (!is.null(mode)) {
+    mode <- rlang::arg_match(mode, c("light", "dark"))
+  }
+
+  if (any(!nzchar(rlang::names2(rlang::list2(...))))) {
+    abort("All arguments in `...` must be named.")
```

Comment:
I think we already `importFrom` for `list2()`, and `arg_match()`/`names2()` is becoming common enough that I'd be inclined to also `importFrom`

Reply:
I never remember what's imported and what isn't. I'm open to importing just these functions, but if we're adding more imported functions from rlang, I'd rather just `@import rlang` -- which also opens us up to using the rlang type checking functions.

Reply:
Yea, I'm OK with that